src/pin.py

```diff
@@ -4,9 +4,12 @@
 from platform import storage_root
 from rng import get_random_bytes
 from ubinascii import hexlify, unhexlify
+from micropython import const
 import ujson as json
+import os
 
 secret_fname = "%s/%s" % (storage_root, "secret.json")
+login_fname = "%s/%s" % (storage_root, "login.json")
```

Comment:
i used a new file for the pin counter, as using secret_fname would require editing it, which doesnt seem optimal to me (e.g. the file could get corrupt?). Let me know if you think otherwise and i will corect it.

Reply:
I think it is a good approach.
Later, when we move the secret key to a fixed location in flash, we should move the PIN counter away from micropython storage as well.